src/SFA.DAS.EmployerIncentives.Commands/UpdateIncentiveApplication/UpdateIncentiveApplicationCommandHandler.cs

```diff
@@ -0,0 +1,56 @@
+﻿using SFA.DAS.EmployerIncentives.Abstractions.Commands;
+using SFA.DAS.EmployerIncentives.Commands.Persistence;
+using SFA.DAS.EmployerIncentives.Domain.Factories;
+using SFA.DAS.EmployerIncentives.Domain.IncentiveApplications;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.EmployerIncentives.Commands.UpdateIncentiveApplication
+{
+    public class UpdateIncentiveApplicationCommandHandler : ICommandHandler<UpdateIncentiveApplicationCommand>
+    {
+        private readonly IIncentiveApplicationFactory _domainFactory;
+        private readonly IIncentiveApplicationDomainRepository _domainRepository;
+
+        public UpdateIncentiveApplicationCommandHandler(IIncentiveApplicationFactory domainFactory, IIncentiveApplicationDomainRepository domainRepository)
+        {
+            _domainFactory = domainFactory;
+            _domainRepository = domainRepository;
+        }
+
+        public async Task Handle(UpdateIncentiveApplicationCommand command, CancellationToken cancellationToken = default)
+        {
+            var application = await _domainRepository.Find(command.IncentiveApplicationId);
+
+            foreach (var apprenticeship in RemovedApprenticeships(command, application))
+            {
+                application.RemoveApprenticeship(apprenticeship);
```

Comment:
Couldn't you just have a method on the domain object like

application.SetApprenticeships(IEnumerable<Apprenticeship> apprenticeships)

where Apprenticeship is a value object and let the domain object decide whether to add or not since there may be additional rules it may want to check.  Aren't all the saves apprenticeships being replaced anyway?